File: src/java_comp/Singleton.java
package java_comp;

public final class Singleton {
    private static final Singleton INSTANCE = new Singleton();
    public String property1;

    private Singleton() {
    }

    public static Singleton getInstance() {
        return INSTANCE;
    }

    public String getInfo() {
        return "SingleInfo";
    }
}

File: src/Singleton.kt
object Singleton {
    var property1: String? = null
    val info: String
        get() = "SingleInfo"
}

/*
class Singleton private constructor() {
    var property1: String? = null
    val info: String
        get() = "SingleInfo"

    companion object {
        val instance = Singleton()
    }
}*/
